activitysim/core/flow.py

```diff
@@ -0,0 +1,802 @@
+        if orig_col_name is None and "od_skims" in local_d:
+            orig_col_name = local_d["od_skims"].orig_key
+        if dest_col_name is None and "od_skims" in local_d:
+            dest_col_name = local_d["od_skims"].dest_key
+        if stop_col_name is None and "dp_skims" in local_d:
+            stop_col_name = local_d["dp_skims"].dest_key
+    local_d = size_terms_on_flow(local_d)
+    size_term_mapping = local_d.get("size_array", {})
+    if "tt" in local_d:
+        aux_vars = local_d["tt"].export_for_numba()
+    else:
+        aux_vars = {}
+    flow = new_flow(
+        spec,
+        extra_vars,
+        orig_col_name,
+        dest_col_name,
+        trace_label,
+        timeframe=timeframe,
+        choosers=choosers,
+        stop_col_name=stop_col_name,
+        parking_col_name=parking_col_name,
+        size_term_mapping=size_term_mapping,
+        interacts=interacts,
+        zone_layer=zone_layer,
+        aux_vars=aux_vars,
+    )
+    flow.tree.aux_vars = aux_vars
+    return flow
+
+
+def should_invalidate_cache_file(cache_filename, *source_filenames):
+    """
+    Check if a cache file should be invalidated.
+
+    It should be invalidated if any source file has a modification time
+    more recent than the cache file modification time.
+
+    Parameters
+    ----------
+    cache_filename : Path-like
+    source_filenames : Collection[Path-like]
+
+    Returns
+    -------
+    bool
+    """
+    try:
+        stat0 = os.stat(cache_filename)
+    except FileNotFoundError:
+        # cache file does not even exist
+        return True
+    for i in source_filenames:
+        stat1 = os.stat(i)
+        if stat0[ST_MTIME] < stat1[ST_MTIME]:
+            return True
+    return False
+
+
+def scan_for_unused_names(tokens):
+    """
+    Scan all spec files to find unused skim variable names.
+
+    Parameters
+    ----------
+    tokens : Collection[str]
+
+    Returns
+    -------
+    Set[str]
+    """
+    configs_dir_list = inject.get_injectable("configs_dir")
+    configs_dir_list = (
+        [configs_dir_list] if isinstance(configs_dir_list, str) else configs_dir_list
+    )
+    assert isinstance(configs_dir_list, list)
+
+    for directory in configs_dir_list:
+        logger.debug(f"scanning for unused skims in {directory}")
+        filenames = glob.glob(os.path.join(directory, "*.csv"))
+        for filename in filenames:
+            with open(filename, "rt") as f:
+                content = f.read()
+            missing_tokens = set()
+            for t in tokens:
+                if t not in content:
+                    missing_tokens.add(t)
+            tokens = missing_tokens
+            if not tokens:
+                return tokens
+    return tokens
+
+
+@inject.injectable(cache=True)
+def skim_dataset_dict(skim_dataset):
+    from .skim_dataset import SkimDataset
+
+    return SkimDataset(skim_dataset)
+
+
+def skims_mapping(
```

Comment:
Yes, the `SkimDataset`/`DatasetWrapper` classes exactly mimic the `SkimDict`/`SkimWrapper` interface, and the network_los.get_skim_dict(...) automatically gives you a SkimDataset instead of a SkimDict when sharrow is enabled. This allows direct calls like `network_los.get_skim_dict(...).lookup(arbitrary_orig, arbitrary_dest, 'SOME_OMX_CORE')`, without the caller needing to even know if sharrow is enabled or not.